### PR TITLE
feat(cli): forbid combining specs and --latest

### PR DESCRIPTION
**Pull request by KSXGitHub**

### CODE REVIEW

File: pkg-manager/plugin-commands-installation/src/update/index.ts [hunk 1]
```diff
@@ -278,7 +278,12 @@ async function update (
   opts: UpdateCommandOptions
 ) {
   if (opts.latest) {
-    const dependenciesWithTags = dependencies.filter((name) => parseWantedDependency(name).pref)
+    const dependenciesWithTags = dependencies.filter((name) => {
+      if (name === '') return false
+      const firstChar = name.charAt(0)
+      if (firstChar !== '@' && !/^[a-z]/i.test(firstChar)) return false
+      return Boolean(parseWantedDependency(name).pref)
+    })
```
Comment (KSXGitHub) [inline pkg-manager/plugin-commands-installation/src/update/index.ts]: (TODO)

```suggestion
    const isPattern = (name: string) => {
      const firstChar = name.charAt(0)
      return firstChar !== '@' && !/^[a-z]/i.test(firstChar)
    }
    const hasSpec = (name: string) => Boolean(parseWantedDependency(name).pref)
    const dependenciesWithTags = dependencies.filter((name) => {
      if (name === '') return false
      if (isPattern(name)) return false
      return hasSpec(name)
    })
```

**Comment by KSXGitHub** [comment]
@zkochan The tests have passed.

**Comment by zkochan** [comment]
I have pushed some improvements.

**Comment by zkochan** [comment]
Usually we avoid adding tests to the pnpm package unless the test verifies some code in the pnpm package itself. The tests in pnpm are heavier than the tests in plugins.

**Review by zkochan — APPROVED** [review]

**Comment by KSXGitHub** [comment]
It looks like the test failures from the last PR returned. I wonder if merging from upstream will fix this problem.

**Comment by KSXGitHub** [comment]
It's weird though, I cannot reproduce the test failures on my machine.

**Pull request merged**